### ISSUE TITLE
GitHub token seems to be required now, with bad error message

### ISSUE THREAD

**Issue by lilyball** [Mic92/nixpkgs-review#105]
When I try to run `nix-review` now I get an HTTP 401 Unauthorized error. I'm not giving nix-review a token, and it should only be accessing public data by default, so I don't know why it's failing. The error is rather obtuse as it's just a very long stack trace but it's definitely coming from the call to the GitHub API.

**Comment by Mic92** [comment]
Do you have by chance an out-of-date token in `cat $HOME/.config/hub`?

**Comment by lilyball** [comment]
Yes I do. I'm rather surprised that nixpkgs-review would be reading .config/hub.

**Comment by lilyball** [comment]
Actually, only one of my computers has an outdated token there. The other computer has a perfectly valid token for a GitHub Enterprise installation and no token for github.com. So it looks like nixpkgs-review isn't validating the domain before it grabs the token.

**Comment by Mic92** [comment]
Yes. I should add that.

**Comment by asymmetric** [comment]
Does it need the token at all?

**Comment by lilyball** [comment]
AFAIK it should only be necessary if you instruct `nixpkgs-review` to leave a comment on the PR. It shouldn't be necessary for the default functionality.

**Comment by Mic92** [comment]
Yes it's optional

**Comment by blaggacao** [comment]
Coming from https://github.com/NixOS/nixpkgs/pull/92612#discussion_r451188524 setting the token on first time usage through prompt would be a good solution.

**Comment by omasanori** [comment]
It would be great to display which scopes the tool needs, otherwise, one (like me) has to guess the correct combination from dozen of checkboxes. I guess it will be especially painful if that is the first time they issue a new token.

**Comment by Mic92** [comment]
@omasanori also discussed here: https://github.com/Mic92/nixpkgs-review/issues/104 You can add it to the readme if you want.

**Comment by omasanori** [comment]
Thank you for the suggestion, @Mic92! Sure, I will take a look at it.

**Comment by tko** [comment]
ping.

This still seems to be the behavior out of the box. Tried running literally the first command in README (I'm new here)
`nix run 'nixpkgs#nixpkgs-review' pr ...`

and got 
```
urllib.error.HTTPError: HTTP Error 401: Unauthorized
```
at the end of a backtrace.

I happened to have a github token for other purposes and so running `GITHUB_TOKEN=... nix run...` worked.

**Comment by Mic92** [comment]
Maybe, I should just removing hub/gh token parsing.

**Comment by teto** [comment]
I stopped exporting GH_TOKEN into my env for various reasons and had the same behavior.

If 

> AFAIK it should only be necessary if you instruct nixpkgs-review to leave a comment on the PR. It shouldn't be necessary for the default functionality.

then the token parsing could be deferred when running those actual commands.

**Comment by Mic92** [comment]
I think `gh` has a subcommand that can be run to provide an Auth token.

**Comment by oxalica** [comment]
Since 3.0.0, #435 makes regular `nixpkgs-review pr` (without posting the result) always need a GitHub token to work, or it fails with a unclear error message complaining about "not a working tree".

```
-> Attempting to fetch eval results from GitHub actions
$ git worktree remove -f /home/<user>/.cache/nixpkgs-review/pr-368839-2/nixpkgs
fatal: '/home/<user>/.cache/nixpkgs-review/pr-368839-2/nixpkgs' is not a working tree
Failed to remove worktree at /home/oxa/.cache/nixpkgs-review/pr-368839-2/nixpkgs. Please remove it manually. Git failed with: 128
Traceback (most recent call last):
[..]
  File "/nix/store/c9m6yd8fg1flz2j5r4bif1ib5j20a0cy-python3-3.12.8/lib/python3.12/urllib/request.py", line 639, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

IIUC, it's because we now use `--eval=ofborg` as default. Could we fallback to local eval when no token is set? Or, at least fail-fast and mention that a token required in this case?
